backend/src/apiserver/server/run_server.go

```diff
@@ -33,6 +33,14 @@ func (s *RunServer) CreateRun(ctx context.Context, request *api.CreateRunRequest
 	if err != nil {
 		return nil, util.Wrap(err, "Validate create run request failed.")
 	}
+	authorized, err := IsAuthorized(s.resourceManager, ctx, request.Run.ResourceReferences)
+	if err != nil {
+		return nil, util.Wrap(err, "Failed to authorize the requests.")
+	}
+	if authorized == false {
```

Comment:
It looks like this path can never be accessed because you return error when authorized==false. As suggested consider not return error from IsAuthorized() and just let caller(in this case, code here) to initiate a bad request a error. Just like what you did in kfamclient.IsAuthorized()

Reply:
Done